### PR TITLE
Rewrite workgroup views from FBV to CBV (#705)

### PR DESCRIPTION
**Pull request by DeKaN**

### CODE REVIEW

**Comment by LegoStormtroopr** [comment]
Looks like there are three tests that are failing with this, but great work so far.

https://travis-ci.org/aristotle-mdr/aristotle-metadata-registry/jobs/281626975

If you want to quickly test them next time, Aristotle uses tox, so you can just test the Workgroup tests like so:
```
ARISTOTLE_DEV_SKIP_MIGRATIONS=1 tox -e  dj1.10-manage-linux-db-sqlite-search-whoosh -- test aristotle_mdr.tests.main.test_workgroups.WorkgroupMemberTests --settings=aristotle_mdr.tests.settings.settings
```

and you might want to check that the tests pass PEP8:
```
tox -e pep8
```

File: aristotle_mdr/views/workgroups.py [hunk 1]
```diff
@@ -1,128 +1,177 @@
...
-    workgroup = get_object_or_404(MDR.Workgroup, pk=iid)
-    if not (workgroup and request.user in workgroup.members):
-        raise PermissionDenied
-
-    if request.method == 'POST':  # If the form has been submitted...
-        workgroup.removeUser(request.user)
+from aristotle_mdr.views.utils import workgroup_item_statuses, paginate_sort_opts
+
+
+class WorkgroupContextMixin:
```
Comment (LegoStormtroopr) [inline aristotle_mdr/views/workgroups.py]: I did a quick google of the error from this test run: https://travis-ci.org/aristotle-mdr/aristotle-metadata-registry/jobs/282746823

If this is changed to a new-style class, like this:
```
class WorkgroupContextMixin(object):
```

It should be ok.

**Comment by LegoStormtroopr** [comment]
This is looking amazing, I've reviewed the test suite and there looks to be only one important test thats failing (the others are due to 2/3 issues)

if you can check why test `aristotle_mdr.tests.main.test_workgroups.WorkgroupMemberTests.test_manager_can_add_or_remove_users` is failing here:
https://travis-ci.org/aristotle-mdr/aristotle-metadata-registry/jobs/282746833

After that it should be good to merge. thanks again :)

File: aristotle_mdr/views/workgroups.py (hunk 1 above)
Comment (DeKaN) [reply aristotle_mdr/views/workgroups.py]: It should be fixed by https://github.com/aristotle-mdr/aristotle-metadata-registry/pull/706/commits/ba179d0ae81fc73d66d922cc3782499f8c41c622
Travis did not restart build after I pushed this commit

**Comment by DeKaN** [comment]
Yes, tests failed for django 1.10, but not for django 1.8. I will try to fix it.

**Comment by LegoStormtroopr** [comment]
Thanks again, I appreciate it!

File: aristotle_mdr/views/workgroups.py (hunk 1 above)
Comment (LegoStormtroopr) [reply aristotle_mdr/views/workgroups.py]: Is it on your develop branch? Thats real odd.

Comment (DeKaN) [reply aristotle_mdr/views/workgroups.py]: Yes, I've pushed commit when first Python 2.7 test failed. But Travis does not discard build for old commit and does not start from beginning.

**Comment by DeKaN** [comment]
@LegoStormtroopr Check passed :)

**Comment by LegoStormtroopr** [comment]
Looks great to me. Thanks!

**Comment by DeKaN** [comment]
@LegoStormtroopr  may you add label `Hacktoberfest ` to this PR as for issue?

**Comment by LegoStormtroopr** [comment]
Of course @DeKaN , thanks again!

**Pull request merged**